jans-auth-server/common/src/main/java/io/jans/as/common/util/RedirectUri.java

```diff
@@ -356,25 +369,20 @@ private String getJweResponse(String nestedJws) throws Exception {
             jwe.setSignedJWTPayload(jwt);
         }
 
-        // Encryption
+        //Encryption
         if (keyEncryptionAlgorithm == KeyEncryptionAlgorithm.RSA_OAEP
                 || keyEncryptionAlgorithm == KeyEncryptionAlgorithm.RSA1_5) {
             PublicKey publicKey = cryptoProvider.getPublicKey(keyId, jsonWebKeys, null);
-
-            if (publicKey != null) {
-                JweEncrypter jweEncrypter = new JweEncrypterImpl(keyEncryptionAlgorithm, blockEncryptionAlgorithm, publicKey);
-                jwe = jweEncrypter.encrypt(jwe);
-            } else {
+            if (publicKey == null) {
                 throw new InvalidJweException("The public key is not valid");
             }
+
+            JweEncrypter jweEncrypter = new JweEncrypterImpl(keyEncryptionAlgorithm, blockEncryptionAlgorithm, publicKey);
+            jwe = jwtForEncrypt.encryptJwe(jwe, jweEncrypter);
```

Comment:
Why are you using jwtForEncrypt? do you really require to extract it outside? I mean, why cannot be jwtEncrypter?

Reply:
jwtEncrypter was deleted from class.